cli/cmd/uninstall.go

```diff
@@ -25,18 +27,34 @@ func newCmdUninstall() *cobra.Command {
 This command provides all Kubernetes namespace-scoped and cluster-scoped resources (e.g services, deployments, RBACs, etc.) necessary to uninstall Linkerd control plane.`,
 		Example: ` linkerd uninstall | kubectl delete -f -`,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return uninstallRunE(cmd.Context())
+
+			k8sAPI, err := k8s.NewAPI(kubeconfigPath, kubeContext, impersonate, impersonateGroup, 0)
+			if err != nil {
+				return err
+			}
+
+			if !force {
+				podsList, err := k8sAPI.CoreV1().Pods("").List(cmd.Context(), metav1.ListOptions{})
+				if err != nil {
+					return err
+				}
+				for _, pod := range podsList.Items {
+					// Skip core control-plane namespace
+					if pod.Namespace != controlPlaneNamespace && healthcheck.ContainsProxy(pod) {
+						return fmt.Errorf("Please uninject proxy containers before uninstalling the control-plane")
```

Comment:
Instead of returning an error here, I think something like the below would be good:

```suggestion
						fmt.Fprintln(os.Stderr, "Please uninject proxy containers before uninstalling the control-plane")
						os.Exit(1)
```

That way, we are not printing the usage text since it's not really a "usage" issue here. We do this in a few other places of the CLI and probably should try to do it more. There is an open issue for it.

With this change:

```
❯ ./bin/go-run cli uninstall
Please uninject proxy containers before uninstalling the control-plane
```

Reply:
Makes sense.